bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -578,20 +578,35 @@ func (r *KubeadmConfigReconciler) joinControlplane(ctx context.Context, scope *S
 // resolveFiles maps .Spec.Files into cloudinit.Files, resolving any object references
 // along the way.
 func (r *KubeadmConfigReconciler) resolveFiles(ctx context.Context, cfg *bootstrapv1.KubeadmConfig, merge ...bootstrapv1.File) ([]bootstrapv1.File, error) {
-	collected := append(cfg.Spec.Files, merge...)
+	collected := make([]bootstrapv1.File, len(cfg.Spec.Files)+len(merge))
```

Comment:
```suggestion
	collected := make([]bootstrapv1.File, 0, len(cfg.Spec.Files)+len(merge))
```

Then we don't need `index`, but we can just `append`

Reply:
not actually merging anything anymore, so got rid of this altogether and switched to append